src/Components/Components/test/RenderTreeDiffBuilderTest.cs

```diff
@@ -1492,6 +1903,207 @@ public void DoesNotInvokeAssignerAgainForRetainedComponents()
             Assert.Empty(referenceFrames);
         }
 
+        [Fact]
```

Comment:
Does it make sense to test something like the frame type changing with the given same key?
For example:
```
element[k1]
element[k2]

component[k2]
component[k1]
```

Also, can multiple key sequences appear on the same component/element at different ranges?
For example a component like:
```
<section>
foreach(var item in FirstList){
  <div @key=@item.Id>
  </div>
}
</section>
<p>Some extra content</p>
<section>
foreach(var item in SecondList){
  <div @key=@item.Id>
  </div>
}
</section>
```

Reply:
> Does it make sense to test something like the frame type changing with the given same key?

Great suggestion. Previously we didn't handle that because it wasn't possible for a `.razor` component. However it is technically now possible when matching on keys. Added support and test.

> Also, can multiple key sequences appear on the same component/element at different ranges?

This isn't an issue, because we only match within the closest container (element or region). In your example the keys won't interfere with each other because each `<section>` is diffed separately.

Reply:
@SteveSandersonMS You're right. My sample is tainted by the `<section>`. What if those were just `<h1>` (and no `</section>`)

Reply:
If they are all in the same container, then by design we will match across all those keys. There's nothing to differentiate different ranges in that case.

Reply:
@SteveSandersonMS Ok. I think its fair to tell people in that case to nest them into a common element. The scenario that I was thinking of was some sort of reporting UI with different views over the same data, but now that I think more of it, you would likely put each of those collections within some container, like a list or a table, so I think its fair to tell people to do it that way.